lib/messagebird.js

```diff
@@ -395,6 +395,308 @@ module.exports = function (accessKey, timeout) {
+       *
+       * @param {String} name
+       * @param {Object} params
+       * @param {Function} callback
+       * @return void
+       */
+      create: function (name, params, callback) {
+        if (typeof params === 'function') {
+          callback = params;
+          params = {};
+        }
+
+        params.name = name;
+
+        httpRequest('POST', '/groups', params, callback);
+      },
+
+      /**
+       * Deletes an existing group. The callback is invoked with an error if
+       * applicable, but the data will never contain anything meaningful as the
+       * API returns an empty response for successful deletes.
+       *
+       * @param {String} id
+       * @param {Function} callback
+       * @return void
+       */
+      delete: function (id, callback) {
+        httpRequest('DELETE', '/groups/' + id, callback);
+      },
+
+      /**
+       * Lists existing groups. Pagination is optional. If a limit is set, an
+       * offset is also required.
+       *
+       * @param {Number} limit
+       * @param {Number} offset
+       * @param {Function} callback
+       * @return void
+       */
+      list: function (limit, offset, callback) {
+        var params = null;
+
+        if (typeof callback === 'function') {
+          params = {
+            limit: limit,
+            offset: offset
+          };
+        } else {
+          callback = limit;
+        }
+
+        httpRequest('GET', '/groups', params, callback);
+      },
+
+      /**
+       * View an existing group.
+       *
+       * @param {String} id
+       * @param {Function} callback
+       * @return void
+       */
+      read: function (id, callback) {
+        httpRequest('GET', '/groups/' + id, callback);
+      },
+
+      /**
+       * Updates an existing contact. Parmas is optional.
+       *
+       * @param {String} id
+       * @param {String} name
+       * @param {Object} params
+       * @param {Function} callback
+       * @return void
+       */
+      update: function (id, name, params, callback) {
+        if (typeof params === 'function') {
+          callback = params;
+          params = {};
+        }
+
+        params.name = name;
+
+        httpRequest('PATCH', '/groups/' + id, params, callback);
+      },
+
+      /**
+       * Adds anywhere from 1 to 50 contacts to a group.
+       *
+       * @param {String} groupId
+       * @param {String[]} contactIds
+       * @param {Function} callback
+       * @return void
+       */
+      addContacts: function (groupId, contactIds, callback) {
+        // We need to make a PUT request with a body formatted like:
+        // `ids[]=contact-id&ids[]=other-contact-id`. The httpRequest method
+        // encodes all request bodies to JSON though.
+        //
+        // Instead, we'll send a GET request and pass a _method=PUT parameter
+        // that will ask the API to handle our request as a PUT. We can then
+        // provide the contact IDs in the query string.
```

Comment:
This is an unfortunate assumption made by `httpRequest`! I think your solution is the best compromise, but perhaps we can consider a refactoring of the request method (perhaps in a different PR, given time constraints).

Something that would allow us to do the following:
```js
var CONTENT_TYPE_FORM_URLENCODED = 'application/x-www-form-urlencoded';

read: function (id, callback) {
  httpRequest({
    method: 'GET',
    path: '/endpoint' + id,
    callback: callback,
    headers: {
      /*
       * Define some default headers in the httpRequest method,
       * then allow them to be overwritten. We could also avoid the 
       * JSON encode unless the content type is set to 'application/json'
       */
      'Content-Type': CONTENT_TYPE_FORM_URLENCODED
    },
  });
},
```

Reply:
True! `httpRequest` can definitely use some improvement. I've opened an issue (https://github.com/messagebird/messagebird-nodejs/issues/21) for this. Thanks!